pyspecProcScripts/integrate_limits.py

```diff
@@ -1,43 +1,132 @@
+        to show diagnostic plots, set `fl` to the figure list; set `fl` to None in
+        order not to see any diagnostic plots
 
-    Returns
-    =======
-    retval: ndarray
-        A len 2 ndarray tuple with the start and stop indeces for the integration
-        bounds.
     """
-    signal_sign = s.C.sum(axis).run(np.real).run(np.sign)
-    temp = abs(s).real * signal_sign
-
-    # pulled from apodization code
-    sigma = nddata(np.linspace(1e-5,1e3,1000),'sigma').set_units('sigma','s')
-    s_avg = s.C.mean_all_but('t2')
-    gaussians = np.exp(-s_avg.C.fromaxis('t2')**2/2/sigma**2)
-    signal_E = (abs(s_avg * gaussians)**2).sum('t2')
-    signal_E /= signal_E.data.max()
-    filter_width = abs(signal_E-1/sqrt(2)).argmin('sigma').item()
+    temp = s.C.mean_all_but(axis)
     if fl is not None:
+        fl.next('integration diagnostic')
         fl.push_marker()
-        fl.next('signal Energy')
+        fl.plot(abs(temp.C.ft('t2'))/abs(temp.C.ft('t2')).max(), alpha=0.6, label='before convolve')
+    Gaussian_Conv = Gaussian
+    Lorentzian_Conv = Lorentzian
+    Lorentz_to_Gauss = Lorentzian_to_Gaussian
+    #{{{ make sure to calculate Lorentzian filter for L-to-G transform
+    if Lorentz_to_Gauss:
+        print("Must set Lorentzian convolve to True for Lorentzian to Gaussian transformation... doing so now")
+        Lorentzian_Conv = True
+        Gaussian_Conv = False
+    #}}}
+    # Calculating matched filter
+    sigma = nddata(np.linspace(1e-10,1e-1,1000),'sigma').set_units('sigma','s')
+    if Gaussian_Conv:
+        gaussians = np.exp(-temp.C.fromaxis(axis)**2/2/sigma**2)
+        signal_E = (abs(temp * gaussians)**2).sum(axis)
+    elif Lorentzian_Conv:
+        lorentzians = np.exp(-abs(temp.C.fromaxis(axis))/sigma)
+        signal_E = (abs(temp * lorentzians)**2).sum(axis)
+    signal_E /= signal_E.data.max()
+    if Gaussian_Conv:
+        filter_width = abs(signal_E-1/sqrt(2)).argmin('sigma').item()
+    elif Lorentzian_Conv:
+        filter_width = abs(signal_E-signal_E.max()/2).argmin('sigma').item()
+    if fl is not None:
+        fl.next('integration diagnostic -- signal Energy')
         fl.plot(signal_E, human_units=False)
         fl.plot(signal_E['sigma':(filter_width,filter_width+1e-6)],'o', human_units=False)
+    logger.info(strm("FILTER WIDTH IS",filter_width))
+    # Generate the appropriate convolution functions
+    Gaussian_func = lambda x,y: exp(-(x**2)/(2.0*(y**2)))
+    Lorentzian_func = lambda x,y: exp(-(abs(x))/y)
+    if Gaussian_Conv:
+        convfunc = Gaussian_func
+    elif Lorentzian_Conv:
+        convfunc = Lorentzian_func
+    if Lorentz_to_Gauss:
+        # not immediately sure why, but I need to
+        # do this in order to get freq limits
+        # around the peak (otherwise they are
+        # around 0)
+        rough_center = abs(temp).C.mean_all_but('t2').argmax('t2').item()
+        temp.setaxis('t2', lambda t: t- rough_center).register_axis({'t2':0})
```

Comment:
I would not do this.  We should be assuming the data has already been properly phased at this point.